File: csharp/Structural/10-Facade/Subsystem/HandymanAssigner.cs
namespace Facade.Subsystem;

public class HandymanAssigner
{
    public string FindQualifiedHandyman(string category)
    {
        Console.WriteLine($"  👷 HandymanAssigner: Finding {category} handyman...");
        return "James Nghipandua";
    }

    public void Assign(string handyman, string jobId)
    {
        Console.WriteLine($"  👷 HandymanAssigner: {handyman} assigned to Job #{jobId}.");
    }
}

File: typescript/Structural/10-Facade/Subsystem/HandymanAssigner.ts
export class HandymanAssigner {
  findQualifiedHandyman(category: string): string {
    console.log(`  👷 HandymanAssigner: Finding ${category} handyman...`);
    return "James Nghipandua";
  }

  assign(handyman: string, jobId: string): void {
    console.log(
      `  👷 HandymanAssigner: ${handyman} assigned to Job #${jobId}.`,
    );
  }
}
